src/privval.rs

```diff
@@ -67,24 +67,48 @@ impl SignableMsg {
                     timestamp: proposal.timestamp.map(Into::into),
                 };
 
-                cp.encode_length_delimited(&mut bytes)?;
+                canonical.encode_length_delimited(&mut bytes)?;
             }
             Self::Vote(vote) => {
-                let cv = proto::types::CanonicalVote {
+                let canonical = proto::types::CanonicalVote {
                     r#type: vote.vote_type.into(),
                     height: vote.height.into(),
                     round: vote.round.value().into(),
                     block_id: vote.block_id.map(Into::into),
                     timestamp: vote.timestamp.map(Into::into),
                     chain_id: chain_id.to_string(),
                 };
-                cv.encode_length_delimited(&mut bytes)?;
+                canonical.encode_length_delimited(&mut bytes)?;
             }
         }
 
         Ok(bytes.into())
     }
 
+    /// Get the bytes representing a vote extension if applicable.
+    pub fn extension_bytes(&self, chain_id: chain::Id) -> Result<Option<Bytes>, EncodeError> {
+        match self {
+            Self::Proposal(_) => Ok(None),
+            Self::Vote(vote) => {
+                // Only sign extension if actually present
```

Comment:
Even if `vote.extension` is empty, but vote extensions are enabled, the signature must be present.